### DISCUSSION TITLE
How to prevent Github from caching images in markdown files

### DISCUSSION THREAD

**Discussion by paramt** [community/community#22283] [New to GitHub]
<p>I’m trying to add an image to a markdown file in my repo. The server updates the image every minute, but the image is cached by Github and therefore does not update in the readme.</p>
<p>I’m sending a cache-control header from a .htaccess file which looks like this:</p>
<pre><code>set Cache-control: private, max-age=0, no-cache
</code></pre>
<p>I have limited understanding of caching, but I’ve noticed that Github Pages doesn’t cache the image when building the site, but it still caches the image in the readme. </p>
<p><a href="http://206.167.183.187/chart.png" rel="nofollow noopener">Original picture (updated once per minute)</a></p>
<p>Checking the Network tab on Chrome shows that the original image is sending this request header:</p>
<pre><code>Cache-Control: no-cache
</code></pre>
<p><a href="https://camo.githubusercontent.com/6cceb82251c9a01d2f2ab51aefe2dccf3bf2a448/687474703a2f2f3230362e3136372e3138332e3138372f63686172742e706e673f32" rel="nofollow noopener">Cached by Github</a><br>
But Github’s cached image is sending this request header:</p>
<pre><code>Cache-Control: public, max-age=31536000
</code></pre>

**Comment by seveas** [comment]
<p>Hi <a class="mention" href="/paramt">@paramt</a> </p>
<p>When I fetch your source image, I get no Cache-Control headers.</p>
<pre><code>$ curl -sv http://206.167.183.187/chart.png -o /dev/null
* Trying 206.167.183.187...
* TCP_NODELAY set
* Connected to 206.167.183.187 (206.167.183.187) port 80 (#0)
&gt; GET /chart.png HTTP/1.1
&gt; Host: 206.167.183.187
&gt; User-Agent: curl/7.54.0
&gt; Accept: */*
&gt;
&lt; HTTP/1.1 200 OK
&lt; Date: Wed, 27 Mar 2019 16:37:41 GMT
&lt; Server: Apache/2.4.29 (Ubuntu)
&lt; Last-Modified: Wed, 27 Mar 2019 16:37:05 GMT
&lt; ETag: "6c4c-585160c4ebb97"
&lt; Accept-Ranges: bytes
&lt; Content-Length: 27724
&lt; Content-Type: image/png
&lt;
{ [2564 bytes data]
* Connection #0 to host 206.167.183.187 left intact
</code></pre>

**Reply by maciejskorski** [reply]
Hit a similar issue when building github docs. The answer here saved me lots of time, thanks!

**Comment by paramt** [comment, marked as answer]
<p>I finally figured it out and it wasn’t a problem with Github, but rather with my response headers (as <a class="mention" href="/seveas">@seveas</a> pointed out). My .htaccess file wasn’t being recognized.</p>
